src/hooks/useLocalizedRoutes/useLocalizedRoutes.tsx

```diff
@@ -20,6 +20,7 @@ export const useLocalizedRoutes = () => {
           // Split id to get path
           const path = id.split('|')[0];
           // Replace path parameters with translations
+          /* eslint-disable-next-line sonarjs/slow-regex */
           const localizedPath: string = path.replaceAll(/{(.*?)}/g, (m: string) => {
```

Comment:
I did not have enough knowledge of this lint rule and should we act on it. 
Try to remove the line and look more info from: https://sonarsource.github.io/rspec/#/rspec/S5852/javascript

Reply:
I am also not really familiar with this, but I asked Perplexity AI about it, and it gave me this explanation. Could any of these options fix the issue to prevent rule disabling?

To address the "slow-regex" issue reported by SonarQube for your JavaScript regular expression, you have a few options to improve its performance and efficiency:

1. Use a non-capturing group:
Replace the capturing group `(.*)` with a non-capturing group `(?:.*)`. This can slightly improve performance:

```javascript
const localizedPath = path.replaceAll(/{(?:.*?)}/g, (m) => {
  const translationKey = m.slice(1, -1);
  return t(translationKey, { lng });
});
```

2. Use a more specific character set:
Instead of using `.*?`, which matches any character, use a more specific character set that excludes curly braces. This can help prevent potential backtracking issues:

```javascript
const localizedPath = path.replaceAll(/{([^{}]*)}/g, (m, translationKey) => {
  return t(translationKey, { lng });
});
```

3. Use `replace` with a callback function instead of `replaceAll`:
While this doesn't change the regex itself, it can be more efficient for older JavaScript engines:

```javascript
const localizedPath = path.replace(/{([^{}]*)}/g, (m, translationKey) => {
  return t(translationKey, { lng });
});
```

4. If possible, consider using a non-regex approach:
If your use case allows, you could split the string and process it without regex:

```javascript
const localizedPath = path.split('{').map((part, index) => {
  if (index === 0) return part;
  const [translationKey, rest] = part.split('}');
  return t(translationKey, { lng }) + rest;
}).join('');
```

These options should help improve the performance of your code and resolve the "slow-regex" warning from SonarQube. Option 2 or 3 are likely the most straightforward and effective solutions for your specific case[1][3][4].

Citations:
[1] https://stackoverflow.com/questions/61478960/sonarqube-showing-regular-expression-denial-of-service-redos
[2] https://community.sonarsource.com/t/well-cached-scans-sometimes-take-a-very-long-time/88228
[3] https://community.sonarsource.com/t/write-efficient-error-free-and-safe-regular-expressions-regex-in-java/28434
[4] https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-regex
[5] https://community.sonarsource.com/t/write-efficient-error-free-and-safe-regular-expressions-in-javascript-and-typescript/47720
[6] https://rules.sonarsource.com/java/type/security%20hotspot/rspec-5852/
[7] https://rules.sonarsource.com/typescript/tag/regex/rspec-6594/
[8] https://rules.sonarsource.com/java/tag/regex/rspec-6353/

Reply:
Thanks for the suggestions. I tried the approach **3.** and for what I tested it seems to work as before.